### PR TITLE
Variable Dimension Index Conversion

### PR DESCRIPTION
**Pull request by sslattery**
This PR adds variable dimensions to the local-global index conversion utility as well as a 2D unit test.

Part of #340. Depends on #349

### CODE REVIEW

**Comment by sslattery** [comment]
@kwitaechong @streeve This one is next for #340

**Review by streeve — APPROVED** [review]

File: cajita/unit_test/tstIndexConversion.hpp [hunk 1]
```diff
@@ -106,8 +106,6 @@ void testConversion( const std::array<bool, 3>& is_dim_periodic )
         } );
 
     // Compare the results.
-    // FIXME Change back to fine grained EXPECT_EQ when passing on all distros
```
Comment (streeve) [inline cajita/unit_test/tstIndexConversion.hpp:109]: Same thing as the `Halo`

Comment (sslattery) [reply cajita/unit_test/tstIndexConversion.hpp:109]: good catch

**Review by kwitaechong — APPROVED** [review]

File: cajita/src/Cajita_IndexConversion.hpp [hunk 2]
```diff
@@ -61,46 +66,47 @@ struct L2G
...
     }
 
-    // Convert local indices to global indices.
-    KOKKOS_INLINE_FUNCTION
-    void operator()( const int li, const int lj, const int lk, int& gi, int& gj,
-                     int& gk ) const
+    // Convert local indices to global indices - 3D.
+    template <std::size_t NSD = num_space_dim>
+    KOKKOS_INLINE_FUNCTION std::enable_if_t<3 == NSD, void>
+    operator()( const int li, const int lj, const int lk, int& gi, int& gj,
```
[the remark this answers is not in the document]
Comment (sslattery) [reply cajita/src/Cajita_IndexConversion.hpp:108]: good idea - I fused these like the other implementation and it definitely reduced code.

**Pull request merged**